src/main/java/org/openx/data/jsonserde/objectinspector/primitive/ParsePrimitiveUtils.java

```diff
@@ -52,12 +52,18 @@ public static long parseLong(String s) {
     public static Timestamp parseTimestamp(String s) {
         Timestamp value;
         //DateISO8601 Format Acceptance
+        String datePattern = "[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]";
+
         if(s.contains("T")) {
             value = new Timestamp(javax.xml.bind.DatatypeConverter.parseDateTime(s).getTimeInMillis());
         }
         else if (s.indexOf(':') > 0) {
             value = Timestamp.valueOf(s);
-        } else if (s.indexOf('.') >= 0) {
+        }
+        else if(s.matches(datePattern)){
+            value = new Timestamp(javax.xml.bind.DatatypeConverter.parseDate(s).getTimeInMillis());
```

Comment:
That's a funky dependency for parsing dates.  I just merged another pull request for parsing timestamps, that can parse using SimpleDateFormat, check it out